src/providers/s3Provider.ts

```diff
@@ -50,8 +51,13 @@ export class S3Provider implements Provider {
     throw new Error('Method not implemented.');
   }
 
-  async readDirectory(path: string): Promise<ReadDirectoryResult | undefined> {
+  async readDirectory(
+    path: string,
+    options?: ReadDirectoryOptions
+  ): Promise<ReadDirectoryResult | undefined> {
     const directories = new Set<string>();
+
+    const listFiles = options !== undefined ? options.listFiles : true;
```

Comment:
This is the sort of logic I believe needs comments so we can understand the principles behind'em